docs/FAQ.md

```diff
@@ -9,7 +9,20 @@ When you run [`arduino-cli board list`][arduino cli board list], your board does
 
 ## What's the FQBN string?
 
-For a deeper understanding of how FQBN works, you should understand the [Arduino platform specification][0].
+FQBN stands for Fully Qualified Board Name. It has the following format:
+`VENDOR:ARCHITECTURE:BOARD_ID[:MENU_ID=OPTION_ID[,MENU2_ID=OPTION_ID ...]]`, with each `MENU_ID=OPTION_ID` being a
+key-value pair configuration. Each field accepts letters (`A-Z` or `a-z`), numbers (`0-9`), underscores (`_`),
+dashes(`-`) and dots(`.`). The special character `=` is accepted in the configuration value. For a deeper understanding
+of how FQBN works, you should understand the [Arduino platform specification][0].
+
+## What fields of the FQBN are optional?
+
+The only mandatory field is `BOARD_ID`. When configs are specified, keys cannot be empty while values can. The following
+FQBN examples are valid:
+
+- `arduino::uno`, `:avr:uno`, `::uno`
```

Comment:
I don't understand this piece, AFAIK the mandatory part is the triple `VENDOR:ARCHITECTURE:BOARD_ID` (not just `BOARD_ID`).
The formula above is correct IMHO:
```
VENDOR:ARCHITECTURE:BOARD_ID[:MENU_ID=OPTION_ID[,MENU2_ID=OPTION_ID ...]]
```
from here we can deduce that the optional part is the MENU_ID=OPTION_ID.
Also, the OPTION_ID must match one of the possible value for the corresponding MENU_ID, it cannot be omitted:

```
$ arduino-cli board details -b arduino:avr:mega:cpu=
Error getting board details: Unknown FQBN: getting build properties for board arduino:avr:mega: invalid value '' for option 'cpu'
$ arduino-cli board details -b arduino:avr:mega:cpu
Error getting board details: Invalid FQBN: invalid config option: cpu
$ arduino-cli board details -b arduino:avr:mega:cpu=atmega2560
[...correct output is displayed...]
```

IMHO we should remove the `## What fields of the FQBN are optional?` section entirely.